prototype_source/semi_structured_sparse.rst

```diff
@@ -0,0 +1,542 @@
+(prototype) Accelelerating BERT with semi-structured (2:4) sparsity
+================================================================
+**Author**: `Jesse Cai <https://github.com/jcaip>`_
+
+Like other forms of sparsity, **semi-structured sparsity** is a model optimization technique that seeks to reduce the memory overhead and latency of a neural network at the expense of some model accuracy.
+It is also known as **fine-grained structured sparsity** or **2:4 structured sparsity**.
+
+Semi-structured sparsity derives its name from its unique sparsity pattern, where n out of every 2n elements are pruned.
+In practice, we most often see n = 2, hence 2:4 sparsity. This sparsity pattern is particularly interesting for two reasons:
```

Comment:
I would collapse this down to something like

Today, we most often see n = 2, hence 2:4 sparsity. 2:4 semi-structured sparsity was designed to be efficiently accelerated on GPUs. In 2021 (?confirm?), NVIDIA introduced hardware support for semi-structured sparsity in their Ampere architecture, and have also released fast sparse kernels via CUTLASS/`cuSPARSELt <https://docs.nvidia.com/cuda/cusparselt/index.html>`_.

The second point isn't saying much unless you go into much more detail on alternative sparsity methods, and why they can compress more.

Reply:
I think it's important to mention that 2:4 sparsity degrades model accuracy less than other sparse methods. 

I'm thinking something like:
```
Today, we most often see n = 2, hence 2:4 sparsity. Semi-structured (2:4) sparsity is particularly interesting because it offers GPU hardware acceleration with minimal loss to model accuracy. In 2020, NVIDIA introduced hardware support for semi-structured sparsity in their Ampere architecture, and have also released fast sparse kernels via CUTLASS/cuSPARSELt <https://docs.nvidia.com/cuda/cusparselt/index.html>_. 
```